### PR TITLE
lanyard/client: add Golang API client for lanyard

### PR DESCRIPTION
**Pull request by worm-emoji**
adds new `clients/go/lanyard` package. 

this package will be imported by code that uses the Lanyard API. 

i also wrote integration tests using this client that only run on pushes to `main` or PRs that change the clients.

### CODE REVIEW

**Review by malonehedges — APPROVED** [review]
nice looks great!

File: clients/go/lanyard/client.go [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+type CreateTreeRequest struct {
+	// UnhashedLeaves is a slice of addresses or ABI encoded types
+	UnhashedLeaves []hexutil.Bytes `json:"unhashedLeaves"`
+
+	// LeafTypeDescriptor describes the abi-encoded types of the leaves, and
+	// is required if leaves are not address types
+	LeafTypeDescriptor []string `json:"leafTypeDescriptor,omitempty"`
+
+	// PackedEncoding is true by default
+	PackedEncoding types.JsonNullBool `json:"packedEncoding,omitempty"` // what's sent over the wire
```
Comment (ryandotsmith) [inline clients/go/lanyard/client.go:123]: What behavior are we looking for with PackedEncoding?

Comment (worm-emoji) [reply clients/go/lanyard/client.go:123]: ah, that comment was old, removed!

Comment (worm-emoji) [reply clients/go/lanyard/client.go:123]: (otherwise just want to maintain same behavior as the http api0

**Review by ryandotsmith — APPROVED** [review]
Let's Get This Merged!

**Pull request merged**